File: implementations/java-springboot/harness/test/testdata/utc-timestamp-source/bad-bare-localdatetime-now/account/domain/Account.java
package com.example.accountservice.account.domain;

import java.time.LocalDateTime;

public class Account {
    private LocalDateTime createdAt;

    private Account() {
    }

    public static Account create(String ownerId) {
        Account account = new Account();
        account.createdAt = LocalDateTime.now();
        return account;
    }
}

File: implementations/kotlin-springboot/harness/test/testdata/aggregate-no-public-setters/good/domain/Account.kt
package com.example.accountservice.account.domain

import java.time.LocalDateTime

class Account private constructor() {
    var accountId: String = ""
        private set

    var status: String = "ACTIVE"
        private set

    var createdAt: LocalDateTime = LocalDateTime.now()
        private set

    companion object {
        fun create(): Account = Account().apply { this.accountId = "acc-1" }
    }

    fun suspend() {
        status = "SUSPENDED"
    }
}
